src/TropicalGeometry/variety_prime.jl

```diff
@@ -0,0 +1,135 @@
+################################################################################
+#
+#  Tropicalization of prime ideals
+#
+#  WARNING: assumes without test that `I` is prime
+#
+################################################################################
+
+function tropical_variety_prime(I::MPolyIdeal, nu::TropicalSemiringMap; weighted_polyhedral_complex_only::Bool=false)
+    # compute a reduced GB to check whether I is homogeneous
+    G = groebner_basis(I,complete_reduction=true)
+
+    if all(Oscar._is_homogeneous.(G))
```

Comment:
This code first computes a `Vector{Bool}` via broadcast and then invokes `all` on it. In particular it allocates and can't do any short circuit.

Better then to use this:
```suggestion
    if all(Oscar._is_homogeneous, G)
```

Here is a toy benchmark demonstrating the difference:
```
julia> v = collect(1:100);

julia> @benchmark all(iseven.(v))
BenchmarkTools.Trial: 10000 samples with 209 evaluations.
 Range (min … max):  360.048 ns … 855.263 ns  ┊ GC (min … max): 0.00% … 0.00%
 Time  (median):     374.402 ns               ┊ GC (median):    0.00%
 Time  (mean ± σ):   378.889 ns ±  17.451 ns  ┊ GC (mean ± σ):  0.00% ± 0.00%

     ▅▆▇▆█▅▄▁▂▁
  ▁▂▅██████████▇▆▅▄▄▄▄▃▃▃▂▂▂▂▂▂▂▂▂▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁ ▃
  360 ns           Histogram: frequency by time          456 ns <

 Memory estimate: 160 bytes, allocs estimate: 4.

julia> @benchmark all(iseven, v)
BenchmarkTools.Trial: 10000 samples with 870 evaluations.
 Range (min … max):  135.345 ns … 245.210 ns  ┊ GC (min … max): 0.00% … 0.00%
 Time  (median):     140.566 ns               ┊ GC (median):    0.00%
 Time  (mean ± σ):   140.761 ns ±   4.574 ns  ┊ GC (mean ± σ):  0.00% ± 0.00%

    ▂▂          ▄█▂  ▁
  ▂▅██▅▅▇█▄▂▂▂▃█████▇█▇▃▂▃▃▃▃▃▄▃▃▂▃▂▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁ ▃
  135 ns           Histogram: frequency by time          156 ns <

 Memory estimate: 0 bytes, allocs estimate: 0.
```

Reply:
Good idea, thanks for the tip!